src/createAnimatedComponent.tsx

```
@@ -750,8 +797,84 @@ export default function createAnimatedComponent(
       return props;
     }
 
+    getSnapshotBeforeUpdate() {
+      if (this._component instanceof HTMLElement) {
+        return this._component.getBoundingClientRect();
+      }
+
+      return null;
+    }
+
+    handleWebAnimation(
+      animationType: LayoutAnimationType,
+      transitionConfig?: TransitionConfig
+    ) {
+      const config =
+        animationType === LayoutAnimationType.ENTERING
+          ? this.props.entering
+          : animationType === LayoutAnimationType.EXITING
+          ? this.props.exiting
+          : animationType === LayoutAnimationType.LAYOUT
+          ? this.props.layout
+          : null;
+
+      if (!config) {
+        return;
+      }
+
+      const initialAnimationName =
+        typeof config === 'function' ? config.name : config.constructor.name;
+
+      // This prevents crashes if we try to set animations that are not defined.
```

How about we give some warning here then, instead of silently failing?